os/hal/ports/SN32/LLD/SN32F24xB/CT/hal_pwm_lld.c

```diff
@@ -0,0 +1,997 @@
+      break;
+    case 3:
+      pwmp->ct->MR3 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM3IOEN_EN;
+      break;
+    case 4:
+      pwmp->ct->MR4 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM4IOEN_EN;
+      break;
+    case 5:
+      pwmp->ct->MR5 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM5IOEN_EN;
+      break;
+    case 6:
+      pwmp->ct->MR6 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM6IOEN_EN;
+      break;
+    case 7:
+      pwmp->ct->MR7 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM7IOEN_EN;
+      break;
+    case 8:
+      pwmp->ct->MR8 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM8IOEN_EN;
+      break;
+    case 9:
+      pwmp->ct->MR9 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM9IOEN_EN;
+      break;
+    case 10:
+      pwmp->ct->MR10 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM10IOEN_EN;
+      break;
+    case 11:
+      pwmp->ct->MR11 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM11IOEN_EN;
+      break;
+    case 12:
+      pwmp->ct->MR12 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM12IOEN_EN;
+      break;
+    case 13:
+      pwmp->ct->MR13 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM13IOEN_EN;
+      break;
+    case 14:
+      pwmp->ct->MR14 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM14IOEN_EN;
+      break;
+    case 15:
+      pwmp->ct->MR15 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM15IOEN_EN;
+      break;
+    case 16:
+      pwmp->ct->MR16 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM16IOEN_EN;
+      break;
+    case 17:
+      pwmp->ct->MR17 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM17IOEN_EN;
+      break;
+    case 18:
+      pwmp->ct->MR18 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM18IOEN_EN;
+      break;
+    case 19:
+      pwmp->ct->MR19 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM19IOEN_EN;
+      break;
+    case 20:
+      pwmp->ct->MR20 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM20IOEN_EN;
+      break;
+    case 21:
+      pwmp->ct->MR21 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM21IOEN_EN;
+      break;
+    case 22:
+      pwmp->ct->MR22 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM22IOEN_EN;
+      break;
+    case 23:
+      pwmp->ct->MR23 = width;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM23IOEN_EN;
+      break;
+    default:
+      ;
+  }
+}
+
+/**
+ * @brief   Disables a PWM channel and its notification.
+ * @pre     The PWM unit must have been activated using @p pwmStart().
+ * @post    The channel is disabled and its output line returned to the
+ *          idle state.
+ * @note    The function has effect at the next cycle start.
+ *
+ * @param[in] pwmp      pointer to a @p PWMDriver object
+ * @param[in] channel   PWM channel identifier (0...channels-1)
+ *
+ * @notapi
+ */
+void pwm_lld_disable_channel(PWMDriver *pwmp, pwmchannel_t channel) {
+
+  switch(channel){
+    case 0:
+      pwmp->ct->IC |= mskCT16_MR0IC;
+      pwmp->ct->PWMIOENB |= mskCT16_PWM0IOEN_DIS;

```

Comment:
This will not work, ORing with 0 (mskCT16_PWM0IOEN_DIS is 0<<0) is a NOP, and will not clear the bit.
I think you mean something like `pwmp->ct->PWMIOENB &= ~mskCT16_PWM0IOEN_EN`

Reply:
nice find. fixed https://github.com/SonixQMK/ChibiOS-Contrib/pull/30/commits/b3c911f27a531e31915b1f263ed0f8396b134f88